Java: update docs on binding transaction to the scope. (#3264)
getsentry/sentry-docs: src/includes/performance/create-transaction-without-binding-to-scope/java.mdx

## Create Transaction Without Binding to The Current Scope

> To avoid passing `ITransaction` objects to method calls when creating child spans, by default started transaction is bound to the current scope.
> 
> When automatic binding is not desired, transactions can be started without binding to the current scope with an overloaded `startTransaction` method. In such case, `Sentry#getSpan` returns `null` and transaction has to be passed to methods manually to create child span:

Reviewer: Do we mean to tell users that the default is to bind a transaction, once started, to the current scope (and combine with the following sentence) - but you can turn this off? If so, suggest: 
```suggestion
By default, a transaction that is started is bound to the current scope. You can avoid passing `ITransaction` objects to method calls when creating child spans without automatically binding to the current scope. In this case, transactions can be started without binding to the current scope with an overloaded `startTransaction` method. When this happens, `Sentry#getSpan` returns `null` and the transaction has to be passed to methods manually to create the child span:
```

Author: I mean that "By default, a transaction that is started is bound to the current scope" - and thanks to this we do not need to pass the `ITransaction` object to method calls.

The alternative is to start transactions with overloaded method taking `bindToScope` parameter with value `false` - but then to create child spans we need to pass it to method calls.

Thanks for your patience! I don't find it easy to express in clear and concise way.

Reviewer 2: @PeloWriter would you have a different suggestion here?

Reviewer: Sorry, @marandaneto - I didn't see that this was awaiting my input again. 

how about:
"By default, a transaction that is started is bound to the current scope. As a result, you do not need to pass the `ITransaction` object to method calls. The alternative is to start transactions with an overloaded `startTransaction` method, setting `bindToScope` to `false. When this happens, `Sentry#getSpan` returns `null` and the transaction has to be passed to methods manually to create the chid spans:"

Author: Since we flipped the flag, this is not applicable anymore.

```java {tabTitle:Java}
import io.sentry.ISpan;
import io.sentry.ITransaction;
import io.sentry.Sentry;

void executeTask() {
  ITransaction transaction = Sentry.startTransaction("transaction-name", "task", false);

  executeSubTask(transaction);

  transaction.finish();
}

private void executeSubTask(ISpan span) {
  ISpan childSpan;
  if (span == null) {
    childSpan = Sentry.startTransaction("transaction-name", "sub-task");
  } else {
    childSpan = span.startChild("operation");
  }
  // ...
  childSpan.finish();
}
```

```kotlin {tabTitle:Kotlin}
import io.sentry.ISpan

> import io.sentry.ITransaction

Reviewer 2: its not being used
